### PR TITLE
refactor: rename `ScopedRef` to `Ref`

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes to #3590

### CODE REVIEW

**Comment by magnus-madsen** [comment]
I want https://github.com/flix/flix/issues/3590 in one go. The steps are just a practical way to do it :)

**Comment by jaschdoc** [comment]
Okay dokay :)

> We take these steps (each in a separate PR):

I think that may have caused the confusion :joy:

**Comment by magnus-madsen** [comment]
> Okay dokay :)
> 
> > We take these steps (each in a separate PR):
> 
> I think that may have caused the confusion 😂

Why did I even write that? :O

**Comment by jaschdoc** [comment]
RFR

File: main/test/flix/Test.Exp.Ref.Ref.flix [hunk 1]
```diff
@@ -1,72 +1,72 @@
 namespace Test/Exp/Ref/Ref {
 
     @test
-    def testRef01(): Ref[Unit] & Impure = ref ()
+    def testRef01(): Ref[Unit, Impure] & Impure = ref ()
```
Comment (magnus-madsen) [inline main/test/flix/Test.Exp.Ref.Ref.flix:4]: Can these be `static` or `Static` if you will?

**Comment by magnus-madsen** [comment]
Nice! That's the easy one down. Next up ScopedArray :)

**Pull request merged**